Domaciukoly20/driving_cars.py

```diff
@@ -0,0 +1,54 @@
+import time
+
+class Car:
+    def __init__(self, economy, color, model, mileage=0, fuel=100, **kwargs):
+        super().__init__(**kwargs)
+        self._mileage = mileage
+        self._fuel = fuel
+        self._economy = economy
+        self._color = color
+        self._model = model
+
+    def drive(self, distance):
+        for i in range(int(distance)):
+            if self._fuel == 0:
+                print("No fuel left!")
+                break
+            elif self._fuel < self._economy:
+                print("Not enough fuel!")
+                self._fuel=0
+                break
+            else:
+                self._fuel -= self._economy
+                time.sleep(1)
+                print("Driving...")
+                break
+        self._mileage += distance
+
+    def distance_left(self):
+        return self._fuel / self._economy
+
+    def fuel_up(self):
+            self._fuel += 20
+
+import random
+models = ['Toyota', 'Honda', 'Ford', 'Chevrolet','Nissan']
+colors = ['blue','green','yellow','black','red','green']
+
+cars = []
+
+for i in range(10):
+    cars.append(Car(random.randrange(100), random.choice(colors), random.choice(models)))
```

Comment:
fuel economy 100? is this a Tank or a plane? :) I'd say 20 - is max here. and 4 is a min (even if the car is hybrid)